orchestration/flows/bl832/tomography_hpc.py

```diff
@@ -0,0 +1,343 @@
+        pass
+
+
+class ALCFTomographyHPCController(TomographyHPCController):
+    """
+    Implementation of TomographyHPCController for ALCF.
+    Methods here leverage Globus Compute for processing tasks.
+
+    Args:
+        TomographyHPCController (_type_): _description_
+    """
+
+    def __init__(self):
+        pass
+
+    def reconstruct(
+        self,
+        file_path: str = "",
+        is_export_control: bool = False,
+        config: Optional[Dict[str, Any]] = None
+    ) -> bool:
+
+        # uses Globus Compute to reconstruct the tomography
+        pass
+
+    def build_multi_resolution(
+        self,
+        file_path: str = "",
+        is_export_control: bool = False,
+        config: Optional[Dict[str, Any]] = None
+    ) -> bool:
+        # uses Globus Compute to build multi-resolution tomography
+        pass
+
+
+class NERSCTomographyHPCController(TomographyHPCController):
+    """
+    Implementation for a NERSC-based tomography HPC controller.
+
+    Submits reconstruction and multi-resolution jobs to NERSC via SFAPI.
+    """
+
+    def __init__(self):
+        self.client = self._create_nersc_client()
+
+    def reconstruct(
+        self,
+        file_path: str = "",
+        is_export_control: bool = False,
+        config: Optional[Dict[str, Any]] = None
+    ) -> bool:
+        """
+        Use NERSC for tomography reconstruction
+        """
+        logger.info("Starting NERSC reconstruction process.")
+
+        if is_export_control:
+            logger.warning("File is export controlled; skipping NERSC reconstruction.")
+            return False
+        else:
+            logger.info("File is not export controlled; proceeding with NERSC reconstruction.")
+
+        if not config:
+            config = Config832()
+
+        nersc_reconstruction_success = self._submit_nersc_reconstruction_job(
+            file_path=file_path
+        )
+
+        logger.info(f"Was NERSC reconstruction successful: {nersc_reconstruction_success}")
+
+        return nersc_reconstruction_success
+
+    def build_multi_resolution(
+        self,
+        file_path: str = "",
+        is_export_control: bool = False,
+        config: Optional[Dict[str, Any]] = None
+    ) -> bool:
+        """Use NERSC to make multiresolution version of tomography results."""
+
+        if is_export_control:
+            logger.warning("File is export controlled; skipping NERSC multi-resolution task.")
+            return False
+        else:
+            logger.info("File is not export controlled; proceeding with NERSC multi-resolution task.")
+
+        if not config:
+            config = Config832()
+
+        nersc_multi_resolution_success = self._submit_nersc_multi_resolution_job(
+            file_path=file_path
+        )
+
+        logger.info(f"Was NERSC multi-resolution conversion successful: {nersc_multi_resolution_success}")
+
+        return nersc_multi_resolution_success
+
+    def _create_nersc_client(self) -> NerscClient:
```

Comment:
This shouldn't be an instance method (self is never used.) I'd rather make this a static method or a separate function so that we can inject the client into this class.